### PR TITLE
Common funcs for filter block indices

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
used for flush/refresh/clear cache

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/utils/constants.ts
Comment (zhichao-aws) [inline public/utils/constants.ts:1]: ReadOnly, MetaData, ReadOnlyAllowDelete will block clear cache; Closed will block all three actions

**Review by gaobinlong — APPROVED** [review]
Looks good to me.

File: public/utils/helpers.ts [hunk 1]
```diff
@@ -45,3 +49,101 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
...
+// import { aliasBlockedPredicate, filterBlockedItems } from "./helpers";
+// import { IndexOpBlocksType } from "./constants";
+// const result = filterBlockedItems<IAlias>(services, selectedItems, IndexOpBlocksType.Closed, aliasBlockedPredicate)
+
+interface BlockedIndices {
+  [indexName: string]: String[];
+}
+
+export async function getBlockedIndices(broserServices: BrowserServices): Promise<BlockedIndices> {
+  interface ClusterBlocksStateResponse {
```
Comment (SuZhou-Joe) [inline public/utils/helpers.ts:64]: Why do we declare a type inside a function? It'd be better to move outside the function.

Comment (zhichao-aws) [reply public/utils/helpers.ts:64]: Sure, I thought this interface is only used by this function. Will move it outside

File: public/utils/helpers.ts [hunk 2]
```diff
@@ -45,3 +49,101 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
...
+export function aliasBlockedPredicate(item: IAlias, blockedItemsSet: Set<String>): boolean {
+  return !!item.indexArray.find((indexName) => blockedItemsSet.has(indexName));
+}
+
+export function dataStreamBlockedPredicate(item: DataStream, blockedItemsSet: Set<String>): boolean {
+  return !!item.indices.find((dataStreamIndex) => blockedItemsSet.has(dataStreamIndex.index_name));
+}
+
+export async function filterBlockedItems<T>(
+  broserServices: BrowserServices,
```
Comment (SuZhou-Joe) [inline public/utils/helpers.ts:119]: typo here, browserServices

File: public/utils/helpers.ts [hunk 3]
```diff
@@ -45,3 +49,101 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
...
+  });
+  return blockedIndices;
+}
+
+export interface FilteredBlockedItems<T> {
+  unBlockedItems: T[];
+  blockedItems: T[];
+}
+
+export function indexBlockedPredicate(item: CatIndex, blockedItemsSet: Set<string>): boolean {
```
Comment (SuZhou-Joe) [inline public/utils/helpers.ts:106]: A parameters with CatIndex is too much for this function, it only consume with the index params here but requires a big set of other useless parameters. The same as IAlias and DataStream, I would recommend using the type: Pick<CatIndex, "index">.

File: public/utils/helpers.ts [hunk 4]
```diff
@@ -45,3 +49,102 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
...
+
+export function aliasBlockedPredicate(item: IAlias, blockedItemsSet: Set<String>): boolean {
+  return !!item.indexArray.find((indexName) => blockedItemsSet.has(indexName));
+}
+
+export function dataStreamBlockedPredicate(item: DataStream, blockedItemsSet: Set<String>): boolean {
+  return !!item.indices.find((dataStreamIndex) => blockedItemsSet.has(dataStreamIndex.index_name));
+}
+
+export async function filterBlockedItems<T>(
```
Comment (SuZhou-Joe) [inline public/utils/helpers.ts:119]: I'd recommend making this function as pure filter function, without the async request to get blocked indexes. The benefit will be that the filter function can be used in other cases, not only with the cluster.state API. But you can make the call.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**